src/convert_json.h

```diff
@@ -21,10 +21,19 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define __CONVERT_JSON_H__
 
 #include "json/json.h"
+#include "exceptions.h"
 
 struct ModStoreMod;
 struct ModStoreModDetails;
 
+class JsonFetchException : public BaseException {
+public:
+	JsonFetchException(const char *s):
+		BaseException(s)
+	{
```

Comment:
Maybe JsonFetchException could have a http_response_code member added to it. Perhaps there could even be a JsonFetchException::getFriendlyErrorString().